src/build-data/makefile.in

```diff
@@ -9,10 +9,10 @@ PYTHON_EXE     = %{python_exe}
 
 ABI_FLAGS      = %{cxx_abi_flags}
 LANG_FLAGS     = %{cc_lang_flags}
-CXXFLAGS       = %{cc_compile_flags}
+CXXFLAGS       = %{cc_sysroot} %{cc_compile_flags}
```

Comment:
Instead of adding it to CXXFLAGS and LDFLAGS (which may be overwritten by user during make invocation), add them to the ABI_FLAGS

ABI_FLAGS      = %{cxx_abi_flags} %{cc_sysroot}